nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -98,6 +99,32 @@ public MongoDBDirectory getDirectory() {
         return (MongoDBDirectory) directory;
     }
 
+    @Override
+    public DocumentModel getEntryFromSource(String id, boolean fetchReferences) throws DirectoryException {
+        String idFieldName = directory.getSchemaFieldMap().get(getIdField()).getName().getPrefixedName();
+        DocumentModelList result = doQuery(Collections.singletonMap(idFieldName, id), Collections.emptySet(),
+                Collections.emptyMap(), true, -1, 0, false);
```

Comment:
Maybe give a `limit == 1` to `doQuery` as we just interested in the first element.